### ISSUE TITLE
Compiling issue

### ISSUE THREAD

**Issue by KormaX-of-DOS** [joncampbell123/dosbox-x#1512]
Hi,

I have problems compiling DOSBox-X with MingW. As the latest version doesn't yet have the full set of binary builds I tried to compile the HX.DOS version myself in order to try it, but the cimpilation always stops with cryptic error messages. I tried various versions of the required libraries, but the lates point I can get is that MAKE -j3 starts, but terminates with Error1. ./configure reports most versions of SDL_net as present but not compilable and explicitely asks me to report it to the author, that's mainly why I am here (as I suppose the other binaries will come soon as in general), but otherwise I am afraid that either the code is heavily tied to specific versions of the libraries or MingW itself that's not mentioned in the README or there are  hidden difficulties that are not trivial for non-professionals, preventing me (and possibly others) to successfully compile the program, but I was going to try a few ./configure switches in order to try stuff not generally built into the DOS builds. A more specific description about which library versions are used in the official binaries would be helpful :) Otherwise I look forward for the official builds as the new version has an improved level of coolness again.

Thank you.

**Comment by Allofich** [comment]
Do you have the latest source code? There was just a Pull Request accepted the other day that is supposed to have fixed MinGW & HX-DOS builds.
https://github.com/joncampbell123/dosbox-x/pull/1509

**Comment by Wengier** [comment]
I fixed building issues in the main DOSBox-X code for MinGW, not in libraries like SDL_net, but he seems to have issues with such libraries. With the fix I successfully built the entire code with MinGW 64-bit.

**Comment by KormaX-of-DOS** [comment]
Thank you guys. I booted into Windows just in order to try this one. :D Seems like I downloaded the program too early this time.

**Comment by KormaX-of-DOS** [comment]
Unforunately it doesn't work. I downloaded the latest master so the fixes are supposed to be included, yet it won't compile. Any advice according to the error messages?

**Comment by KormaX-of-DOS** [comment]
Unfortunately it doesn't work. I downloaded the latest master so the fixes are supposed to be included, yet it won't compile. It says Error 1 at sdlmain.o. Any advice?

**Comment by Wengier** [comment]
@KormaX-of-DOS Can you post the full error messages?

**Comment by KormaX-of-DOS** [comment]
Here's a screenshot

**Comment by KormaX-of-DOS** [comment]
Here.

**Comment by Wengier** [comment]
I don't see anything above except "Here."

**Comment by KormaX-of-DOS** [comment]
Here again. It didn't attach my screen capture for some reason.

**Comment by joncampbell123** [comment]
@KormaX-of-DOS Is Github preventing you from attaching screenshots for some reason?

**Comment by KormaX-of-DOS** [comment]
Yes, my screenshot wasn't attached for some reason, not from DOS and not from my phone. You reacted more quickly than I was able to solve the problem. I uploaded the file to 0x0.st:

http://0x0.st/i2Rv.png

Thanks for your help. :)

**Comment by joncampbell123** [comment]
Good screenshot, however all the compiler warnings caused the error to scroll completely off the terminal window. You'll need to scroll back to where the error lines appear.

Also at least here in Linux with Firefox I am able to click the "Attach files by..." region and upload PNGs. What does Github do on your end if you do that?

**Comment by KormaX-of-DOS** [comment]
Theoretically the same is true here, but eventually the file is not uploaded. It says I can upload GIF, JPEG, JPG, PNG, DOCX, GZ, LOG, PDF, PPTX, TXT, XLSX or ZIP files. But on DOS a message is shown that says I need a permission to upload documents. But it doesn't work on my phone either. And I made the screen capture on Windows 7 and wanted to upload it immediately an it didn't work on that OS either. I will try building the program again but I'll redirect the output into a file and send it here, but I was too unfamiliar with SH so I didn't it earlier. Does the HX.DOS version or the Windows 32 bit MingW version compile for you without problem?

**Comment by Wengier** [comment]
I compiled MinGW & HX-DOS versions in 64-bit MinGW version, not 32-bit. In 64-bit MinGW version the code built successfully for me. In fact, I am now using 64-bit MinGW as my main testing platform in Windows (instead of Visual Studio) because it compiles faster than Visual Studio.

**Comment by joncampbell123** [comment]
The issue with VS2019 seems to be the linker and "whole program optimization". In the task manager the linker looks normal until at some point memory usage rises dramatically up to 4GB, then it gives up.

Turning off whole program optimization just makes the compiler complain about heap space on sdlmain.cpp.

**Comment by KormaX-of-DOS** [comment]
So I should modify CFLAGS in the generated Makefile to contain -O0?

**Comment by Wengier** [comment]
@KormaX-of-DOS You can try various options of course, but if you are able to upload a screenshot with full error lines shown, then we may be able to help directly too.

**Comment by KormaX-of-DOS** [comment]
Hello!

I don't actually understand what you mean by full error lines. There are a few warnings and then what I had shown .The thing is that after I execute ./build-mingw-hx-dos it terminates saying no child processes then I need to run ./configure and make -j3 manually. So I don't know where the problem might be. I took more screenshots this time. I hope there will be a solution because I would like to use the program.

http://0x0.st/iL7R.jpg http://0x0.st/iL77.jpg http://0x0.st/iL7h.jpg

**Comment by Wengier** [comment]
@KormaX-of-DOS Some important new features such as FAT32 support have been added to the recent code (and we are still doing some finish-ups). So perhaps you can wait for the next version will may come with an official HX-DOS binary.

**Comment by KormaX-of-DOS** [comment]
Hello!
I downloaded the new version and it is fantastic. Also thanks for the printer emulation in the DOS build.

It was worth waiting, thank you for your help and work.

Regards:
KormaX
